include/mesh/mesh_function.h

```diff
@@ -146,27 +146,45 @@ class MeshFunction : public FunctionBase<Number>,
 
   /**
    * Computes values at coordinate \p p and for time \p time, which
-   * defaults to zero.
+   * defaults to zero, optionally restricting the point to the passed
+   * subdomain_ids. This is useful in cases where there are multiple
+   * dimensioned elements, for example.
    */
   void operator() (const Point& p,
                    const Real time,
                    DenseVector<Number>& output);
 
   /**
+   * Computes values at coordinate \p p and for time \p time,
+   * restricting the point to the passed subdomain_ids. This is useful in
+   * cases where there are multiple dimensioned elements, for example.
+   */
+  void operator() (const Point& p,
+                   const Real time,
+                   DenseVector<Number>& output,
+                   const std::set<subdomain_id_type>* subdomain_ids);
+
+  /**
    * Computes gradients at coordinate \p p and for time \p time, which
```

Comment:
> No, I'm an idiot. We don't want to force people subclassing FunctionBase to change their code, so we can't change the method signature, so we have to add a new overload.

You want me to go ahead and do that here?

> I don't think we should deprecate the old method, though. Just because people are now allowed to worry about which subdomain they're querying, that doesn't mean they should be forced to worry about it. Ideally, most Functions should still be functions. ;-)

Agreed, I was just thinking so that we could, in the future, have one method were we default to NULL for the subdomain ids instead of persisting with two.